cmake/Version.cmake

```diff
@@ -6,22 +6,62 @@
 #
 #
 
+include(cmake/FileUtil.cmake)
+
 string(TIMESTAMP current_year "%Y")
 set(O3DE_COPYRIGHT_YEAR ${current_year} CACHE STRING "Open 3D Engine's copyright year")
-set(O3DE_VERSION_STRING "0.0.0" CACHE STRING "Open 3D Engine's version")
-set(O3DE_DISPLAY_VERSION_STRING "00.00" CACHE STRING "Open 3D Engine's display version")
-set(O3DE_BUILD_VERSION 0 CACHE STRING "Open 3D Engine's build number")
-set(O3DE_ENGINE_NAME "o3de" CACHE STRING "Open 3D Engine's engine name")
-
-# Optional environment overrides
-if(NOT "$ENV{O3DE_VERSION}" STREQUAL "")
-    set(O3DE_VERSION_STRING "$ENV{O3DE_VERSION}")
-endif()
-
-if(NOT "$ENV{O3DE_DISPLAY_VERSION}" STREQUAL "")
-    set(O3DE_DISPLAY_VERSION_STRING "$ENV{O3DE_DISPLAY_VERSION}")
-endif()
-
-if(NOT "$ENV{O3DE_BUILD_VERSION}" STREQUAL "")
-    set(O3DE_BUILD_VERSION "$ENV{O3DE_BUILD_VERSION}")
-endif()
+
+get_filename_component(ENGINE_JSON_PATH "${CMAKE_CURRENT_LIST_DIR}/../engine.json" ABSOLUTE)
```

Comment:
It would be better to use `${LY_ROOT_FOLDER}/engine.json` here to locate the engine.json

Reply:
I was using that originally but noticed it gets set later on `GeneralSettings.cmake` and wasn't sure if we wanted to include that so early or if that might mess up project based vs engine based setups.  I'll take a closer look at that and do some testing.